rust/lance/src/index/vector/builder.rs

```diff
@@ -1163,7 +1167,12 @@ impl<S: IvfSubIndex + 'static, Q: Quantization + 'static> IvfIndexBuilder<S, Q>
         ivf: &IvfModel,
         reader: &dyn ShuffleReader,
         existing_indices: &[Arc<dyn VectorIndex>],
+        num_indices_to_merge: Option<usize>,
```

Comment:
This API change really surprised me. If we just input an `Option` and return `Ok(None)` if it's some, how about we do this check outside?